pkg/api/types.go

```diff
@@ -44,13 +44,21 @@ type DeschedulerStrategy struct {
 	Params *StrategyParameters
 }
 
+// Namespaces carries a list of included/excluded namespaces
+// for which a given strategy is applicable
+type Namespaces struct {
+	Include []string
+	Exclude []string
+}
+
 // Only one of its members may be specified
 type StrategyParameters struct {
 	NodeResourceUtilizationThresholds *NodeResourceUtilizationThresholds
 	NodeAffinityType                  []string
 	PodsHavingTooManyRestarts         *PodsHavingTooManyRestarts
 	MaxPodLifeTimeSeconds             *uint
 	RemoveDuplicates                  *RemoveDuplicates
+	Namespaces                        Namespaces
```

Comment:
As `Namespaces` is not a new strategy, it's a bit inappropriate to place it inside `StrategyParameters`. Besides, as the comment writes, we should only specify one of these params, but in this case we may use both `Namespaces` and `RemoveDuplicates`(or other strategies). I think we should place `Namespaces` inside [DeschedulerStrategy](https://github.com/kubernetes-sigs/descheduler/blob/07f476dfc4d40207312b7d9e99c0f2c3a7394faf/pkg/api/types.go#L36).

Reply:
`Namespaces` is just another parameter for a strategy so it belongs to `StrategyParameters`. Also, it's up to every strategy to decide which parameters it will read.

Reply:
> Also, it's up to every strategy to decide which parameters it will read.

I agree with this, but I think that actually makes a case to put `Namespaces` in each strategy's struct (eg, [`RemoveDuplicates.Namespaces`](https://github.com/kubernetes-sigs/descheduler/blob/07f476dfc4d40207312b7d9e99c0f2c3a7394faf/pkg/api/types.go#L70-L72) for the strategies that will implement it.

For Namespaces, I think that's every strategy right? I know for pod selector we discussed that some strategies shouldn't implement it, in which case it shouldn't be available as an api option.

Also, I don't think we do anything to strictly enforce that only one of these StrategyParameters may be specified besides convention. Tbh in the next version of this api I would prefer to see `StrategyParameters` replaced with an interface or `object.Runtime` to get rid of this big list (similar to how the scheduler configures [`PluginArgs`](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/types.go#L236)

Reply:
Yeah, `// Only one of its members may be specified` comment might be no longer valid here.

Reply:
> I agree with this, but I think that actually makes a case to put Namespaces in each strategy's struct (eg, RemoveDuplicates.Namespaces for the strategies that will implement it.

Done.